contracts/rewards/src/commands.rs

```diff
@@ -124,12 +124,13 @@ pub fn distribute_reward(
         return Err(ContractError::Unauthorized {});
     }
 
+    let total_distribution_amount: Uint128 = distributions.iter().map(|d| d.amount).sum();
+    if config.spend_limit < total_distribution_amount {
```

Comment:
if someone calls the `distribute_reward` multiple times, it's still possible to transfer more than the specified `spend_limit`.

So the spend limit only limits how much can be sent in a single TX.

That's ok imo, as only whitelisted addresses can invoke a disribution, so this check prevents us from accidently sending too many tokens.